types/tx_msg_v2.go

```diff
@@ -0,0 +1,49 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	protov2 "google.golang.org/protobuf/proto"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+)
+
+type (
+	// MsgV2 defines the interface a transaction message needed to fulfill.
+	MsgV2 = protov2.Message
```

Comment:
This doesn't feel right. Should only exists one struct that represents a 'message' and then an adapter that takes that struct as input and converts it to the desire type (protov1 or protov2)? So we can move the protov1 vs v2 drama to that adapter only ?